contracts/test/ocr2/aggregator.test.ts

```diff
@@ -257,70 +367,109 @@ describe('Aggregator', function () {
-
-      let oracle = oracles[0]
-      let payee = oracle.transmitter
-      let { response: owed } = await aggregator.call('owed_payment', {
-        transmitter: oracle.transmitter.starknetContract.address,
+      await owner.execute({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'set_payees',
+        calldata: CallData.compile({
+          payees: oracles.map((oracle) => ({
+            transmitter: oracle.transmitter.address,
+            payee: oracle.transmitter.address, // reusing transmitter acocunts as payees for simplicity
+          })),
+        }),
       })
+
       // several rounds happened so we are owed payment
-      assert.ok(owed > 0)
+      const payee = oracles[0].transmitter
+      const [owed1] = await provider.callContract({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'owed_payment',
+        calldata: CallData.compile({
+          transmitter: payee.address,
+        }),
+      })
+      assert.ok(parseInt(owed1, 16) > 0)
 
       const availableToValue = ([is_negative, abs_difference]: [boolean, bigint]): bigint => {
         return is_negative ? -abs_difference : abs_difference
       }
 
       // no funds on contract, so no LINK available for payment
-      let { response: available } = await aggregator.call('link_available_for_payment')
-      assert.ok(availableToValue(available) < 0) // should be negative: we owe payments
+      let [isNegative, absDiff] = await provider.callContract({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'link_available_for_payment',
+      })
+      assert.ok(availableToValue([isNegative === '0x1', BigInt(absDiff)]) < 0) // should be negative: we owe payments
 
       // deposit LINK to contract
-      await owner.invoke(token, 'transfer', {
-        recipient: aggregator.address,
-        amount: 100_000_000_000,
+      await owner.execute({
+        contractAddress: token.deploy.address,
+        entrypoint: 'transfer',
+        calldata: CallData.compile({
+          recipient: aggregator.deploy.address,
+          amount: cairo.uint256(100_000_000_000n),
+        }),
       })
 
       // we have enough funds available now
-      available = (await aggregator.call('link_available_for_payment')).response
-      assert.ok(availableToValue(available) > 0)
+      ;[isNegative, absDiff] = await provider.callContract({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'link_available_for_payment',
+      })
+      assert.ok(availableToValue([isNegative === '0x1', BigInt(absDiff)]) > 0)
 
       // attempt to withdraw the payment
-      await payee.invoke(aggregator, 'withdraw_payment', {
-        transmitter: oracle.transmitter.starknetContract.address,
+      await payee.execute({
+        contractAddress: aggregator.deploy.address,
+        entrypoint: 'withdraw_payment',
+        calldata: CallData.compile({
+          transmitter: payee.address,
+        }),
       })
```

Comment:
Nit: rather than calling `CallData.compile` and providing the entrypoint manually, you should be able to use the `populate` method on the contracts:

```ts
await payee.execute(aggregator.populate('withdraw_payment', { transmitter: payee.address }));
```

Reply:
Sounds good - I've replaced all occurrences of `CallData.compile` with `populate`